reconcile/terraform_vpc_peerings.py

```diff
@@ -655,30 +684,71 @@ def run(
     if any(errors):
         sys.exit(1)
 
-    defer(tf.cleanup)
+    runner_params: RunnerParams = dict(
+        tf=tf,
+        dry_run=dry_run,
+        enable_deletion=enable_deletion,
+        defer=defer,
+    )
+
+    if enable_extended_early_exit and get_feature_toggle_state(
+        "terraform-vpc-peerings-extended-early-exit",
+        default=True,
+    ):
+        cache_source = CacheSource(terraform_configurations=terraform_configurations)
+        extended_early_exit_run(
+            integration=QONTRACT_INTEGRATION,
+            integration_version=QONTRACT_INTEGRATION_VERSION,
+            dry_run=dry_run,
+            cache_source=cache_source,
+            shard=account_name if account_name else "",
+            ttl_seconds=extended_early_exit_cache_ttl_seconds,
+            logger=logging.getLogger(),
+            runner=runner,
+            runner_params=runner_params,
+            log_cached_log_output=log_cached_log_output,
+        )
+    else:
+        runner(**runner_params)
+
+
+class CacheSource(TypedDict):
+    terraform_configurations: dict[str, str]
+
+
+class RunnerParams(TypedDict):
+    tf: terraform.TerraformClient
+    dry_run: bool
+    enable_deletion: bool
+    defer: Callable | None
+
+
+def runner(
+    dry_run: bool,
+    tf: terraform.TerraformClient,
+    enable_deletion: bool = False,
+    defer: Callable | None = None,
+) -> ExtendedEarlyExitRunnerResult:
+    if defer:
+        defer(tf.cleanup)
```

Comment:
`defer` should be called right after `tf` https://github.com/app-sre/qontract-reconcile/pull/4381/files#diff-79f381a7c684c7a6788d3410826abaa6412cd5767ee6694e8e06154fe8e0268fR674-R682, then it's not needed to be part of `RunnerParams`

Reply:
Fixed in [daa3332](https://github.com/app-sre/qontract-reconcile/pull/4381/commits/daa3332b0ef1a46304faafb082ce85dd382be201)